webpack/scenes/ContentViews/Copy/__tests__/copyContentView.test.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { renderWithRedux, patientlyWaitFor, fireEvent } from 'react-testing-lib-wrapper';
+
+import { nockInstance, assertNockRequest } from '../../../../test-utils/nockWrapper';
+import api from '../../../../services/api';
+import CopyContentViewForm from '../CopyContentViewForm';
+
+const cvCopyData = require('./contentViewCopyResult.fixtures.json');
+
+const cvId = '1';
+const cvCopyPath = api.getApiUrl(`/content_views/${cvId}/copy`);
+const setModalOpen = jest.fn();
+
+const copyParams = {
+  id: cvId,
+  name: 'cv copy',
+};
+
+const copiedCVDetails = { ...cvCopyData };
+
+const form = <CopyContentViewForm cvId={cvId} setModalOpen={setModalOpen} />;
+
+test('Can copy content view from form', async (done) => {
+  const copyscope = nockInstance
+    .post(cvCopyPath, copyParams)
+    .reply(201, copiedCVDetails);
+  const { queryByText, getByLabelText } = renderWithRedux(form);
+  expect(queryByText('Name')).toBeInTheDocument();
+
+  fireEvent.change(getByLabelText('input_name'), { target: { value: 'cv copy' } });
+
+  getByLabelText('copy_content_view').click();
+
+  assertNockRequest(copyscope, done);
+});
+
+test('Form closes itself upon save', async (done) => {
+  const copyscope = nockInstance
+    .post(cvCopyPath, copyParams)
+    .reply(201, copiedCVDetails);
+  const { getByText, queryByText, getByLabelText } = renderWithRedux(form);
+  expect(getByText('Name')).toBeInTheDocument();
+
+  fireEvent.change(getByLabelText('input_name'), { target: { value: 'cv copy' } });
+
+  getByLabelText('copy_content_view').click();
+  // Form closes it self on success
+  await patientlyWaitFor(() => {
+    expect(queryByText('Name')).not.toBeInTheDocument();
+  });
+
+  assertNockRequest(copyscope, done);
+});
```

Comment:
One thing I just noticed - this test is pretty similar to the other one, I'm fine with this test being the only test and checking multiple things during the test (the api call is made and the form is closed), these longer tests make sense in functional testing IMO

Reply:
Updated..